src/SvgStream.h

```diff
@@ -18,7 +18,7 @@ namespace svglite { namespace internal {
 template <typename T>
 void write_double(T& stream, double data) {
   std::streamsize prec = stream.precision();
-  uint8_t newprec = std::fabs(data) >= 1 || data == 0. ? prec : std::ceil(-std::log10(std::fabs(data))) + 1;
+  int newprec = std::fabs(data) >= 1 || data == 0. ? prec : std::ceil(-std::log10(std::fabs(data))) + 1;
```

Comment:
Is `std::ceil()` guaranteed to be in `uint8_t` range? If so maybe maybe something like this is preferable?

```suggestion
  uint8_t newprec = std::fabs(data) >= 1 || data == 0. ? prec : std::ceil(-std::log10(std::fabs(data)));
  newprec = newprec == UINT8_MAX ? UINT8_MAX : newprec - 1;
```

Reply:
True, matching the `std::setprecision()` parameter sounds like the best choice to me

https://en.cppreference.com/w/cpp/io/manip/setprecision